### PR TITLE
Fix AssemblyTests on Windows

### PR DESCRIPTION
**Pull request by lefou**
Depends on #3285 

Windows has an env var called `PATHEXT` whish is usually something like this: `.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC`  
Basically a list of extensions considered as executable.  
So the file in Mill's case needs to be a `.bat` file.

Issue analyzed and fixed by @sake92 

Fix https://github.com/com-lihaoyi/mill/issues/3283

### CODE REVIEW

**Review by lefou — COMMENTED** [review]
@sake92 Thank you for the fix! Can you please format it and also bump the Mill version to `0.11.9` so we see it succeeds? Thank you

**Comment by sake92** [comment]
@lefou where do I bump the mill version, in `.config\mill-version` ?

**Comment by lefou** [comment]
> @lefou where do I bump the mill version, in `.config\mill-version` ?

Yes.

**Comment by sake92** [comment]
Seems like scalalib tests are green now.  
https://github.com/com-lihaoyi/mill/actions/runs/10060348007

**Comment by lefou** [comment]
Looks like there are some new unrelated test failures for Windows now. Since we now know, the assembly tests are fixed by this PR, we could remove the Mill bump,  and merge this PR (and close the bounty, I guess).

File: .config/mill-version [hunk 1]
```diff
@@ -1 +1 @@
-0.11.7-17-17dea9
+0.11.9
```
Comment (lefou) [inline .config/mill-version:1]: Try to avoid the possibly unrelated test errors.

```suggestion
0.11.8
```

**Comment by sake92** [comment]
> Looks like there are some new unrelated test failures for Windows now. Since we now know, the assembly tests are fixed by this PR, we could remove the Mill bump, and merge this PR (and close the bounty, I guess).

I guess it would be fair to make the CI green first, before closing the bounty. 😁

**Comment by lefou** [comment]
> > Looks like there are some new unrelated test failures for Windows now. Since we now know, the assembly tests are fixed by this PR, we could remove the Mill bump, and merge this PR (and close the bounty, I guess).
> 
> I guess it would be fair to make the CI green first, before closing the bounty. 😁

Yeah. That was the idea by not bumping Mill to cause unrelated issues.

**Comment by lefou** [comment]
I think the other test failure are valid but unrelated. They probably went under the radar since our Windows CI wasn't fully functional before #3285 .

**Comment by sake92** [comment]
@lefou we can merge this one?  
It is easier to fix them one by one I think.

**Pull request merged**